pkgs/applications/science/biology/spades/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchurl, zlib, bzip2, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "SPAdes";
+  version = "3.13.1";
+
+  src = fetchurl {
+    url = "http://cab.spbu.ru/files/release${version}/${pname}-${version}.tar.gz";
+    sha256 = "0giayz197lmq2108filkn9izma3i803sb3iskv9hs5snzdr9p8ld";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ zlib bzip2 ];
+
+  doCheck = true;
+
+  postUnpack = ''
+    sourceRoot=''${sourceRoot}/src
```

Comment:
@bzizou can you please explain what you're doing with this, instead of just setting `sourceRoot` in the root of the derivation itself?

Reply:
Like that?
`sourceRoot="${pname}-${version}/src";`
It works, but we have to literally define it. Is there a simpler way to set sourceRoot as `<guessed sourceRoot>/src`?